pkg/codegen/dotnet/gen.go

```diff
@@ -114,11 +114,17 @@ func namespaceName(namespaces map[string]string, name string) string {
 	if ns, ok := namespaces[name]; ok {
 		return ns
 	}
-	names := strings.Split(name, "-")
-	for i, name := range names {
-		names[i] = Title(name)
+
+	// name could be a qualified module name so first split on /
+	parts := strings.Split(name, "/")
```

Comment:
There's a `const` `tokens.QNameDelimiter` in `https://github.com/pulumi/pulumi/blob/master/sdk/go/common/tokens/names.go` why not using this here instead of a magic string?